### PR TITLE
 fix: Use ember-power-select for basic event details dropdown

### LINKED ISSUE #4546: Event Creation: form filling error
This is a minor error happened during the event creation process

**To reproduce**
1. Go to 'Edit Event' on 'https://eventyay.com/e/8fa7fd14'
2. Click on 'Basic Details'
3. Scroll down to 'Event Topic' 

**Selection was made and reflected on the live page**
Screenshot from 2020-07-11 12-19-26

**But it did not show on the field in the next visit** 
Screenshot from 2020-07-11 08-53-05

### Desired Fixes
- The event topic field is saved
- Possibility to leave this field blank
- Possibility to insert a custom topic

### PR DESCRIPTION
**Pull request by Haider8**
Fixes #4546 

#### Short description of what this resolves:
There is a library error which becomes visible when there's `&` in the options like in `Science & Technology`.

#### Changes proposed in this pull request:
Use new library `ember-power-select` for dropdowns.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/components/forms/wizard/basic-details-step.js [hunk 1]
```diff
@@ -109,6 +109,13 @@ export default Component.extend(FormMixin, EventWizardMixin, {
     return this.data.event.topic.get('subTopics');
   }),
 
+  topics: computed('this.data.topics', function() {
+    let topic_names = new Array();
+    const topics = this.data.topics.toArray();
+    topics.forEach(topic => topic_names.push(topic.name));
+    return topic_names;
```
Comment (iamareebjamal) [inline app/components/forms/wizard/basic-details-step.js:116]:
```suggestion
    return this.data.topics.toArray().map(topic => topic.name);
```

File: app/templates/components/forms/wizard/basic-details-step.hbs [hunk 2]
```diff
@@ -622,11 +622,12 @@
...
   <div class="field">
+    <label>{{t 'Event Topic'}}</label>
     <PowerSelect
       @selected={{this.data.event.topic.name}}
-      @options={{this.data.topics}}
-      @onChange={{action (mut this.data.event.topic.name)}}
-      @placeholder={{t 'Select Event Topic'}} as |topic|>
+      @options={{this.topics}}
+      @placeholder={{t 'Select Event Topic'}}
+      @onChange={{action (mut this.data.event.topic.name) as |topic|}}>
```
Comment (iamareebjamal) [inline app/templates/components/forms/wizard/basic-details-step.hbs:630]: We don't want to change the name of the topic

Comment (iamareebjamal) [reply app/templates/components/forms/wizard/basic-details-step.hbs:630]: Loop over topic and render topic.name and mutate topic onChange

**Comment by iamareebjamal** [comment]
Is it working now?

**Comment by Haider8** [comment]
Yes

**Comment by iamareebjamal** [comment]
Replace event type topic and subtopic with power select

**Comment by iamareebjamal** [comment]
Remove WIP tag once done with PRs

**Pull request merged**